db/version_set.cc

```diff
@@ -560,12 +580,21 @@ class LevelIterator final : public InternalIterator {
       smallest_compaction_key = (*compaction_boundaries_)[file_index_].smallest;
       largest_compaction_key = (*compaction_boundaries_)[file_index_].largest;
     }
+    hint_within_lower_bound_ =
+        read_options_.iterate_lower_bound != nullptr &&
+        user_comparator_.Compare(ExtractUserKey(file_smallest_key(file_index_)),
+                                 *read_options_.iterate_lower_bound) >= 0;
+    bool hint_within_upper_bound =
+        read_options_.iterate_upper_bound != nullptr &&
+        user_comparator_.Compare(ExtractUserKey(file_largest_key(file_index_)),
+                                 *read_options_.iterate_upper_bound) < 0;
     return table_cache_->NewIterator(
         read_options_, env_options_, icomparator_, *file_meta.file_metadata,
         range_del_agg_, prefix_extractor_,
-        nullptr /* don't need reference to table */,
-        file_read_hist_, for_compaction_, nullptr /* arena */, skip_filters_,
-        level_, smallest_compaction_key, largest_compaction_key);
+        nullptr /* don't need reference to table */, file_read_hist_,
+        for_compaction_, nullptr /* arena */, skip_filters_, level_,
+        smallest_compaction_key, largest_compaction_key,
+        hint_within_upper_bound);
```

Comment:
This is a little bit tricky, for the reason mentioned above. Do we have to pass it in? Can we always check the hint while moving to a new block?

Reply:
okay, will update it to not passing it down.